prompt-service/src/unstract/prompt_service/main.py

```diff
@@ -693,29 +693,51 @@ def run_retrieval(  # type:ignore
     retrieval_type: str,
     metadata: dict[str, Any],
 ) -> tuple[str, str]:
+    context: str = ""
     prompt = output[PSKeys.PROMPTX]
     if retrieval_type == PSKeys.SUBQUESTION:
-        subq_prompt = (
-            f"Generate a sub-question from the following verbose prompt that will"
-            f" help extract relevant documents from a vector store:\n\n{prompt}"
+        subq_prompt: str = (
+            f"I am sending you a verbose prompt {prompt} . "
```

Comment:
Better to separate the prompt using `\n\n` at the start and end.

Reply:
How will the preceding `\n` help here?

Reply:
Spacing makes it better for LLM to understand it better. We need to keep the prompt seperately apart from the system prompt like this.

```I am sending you a verbose prompt:

{prompt}

....
```

Reply:
Get that, addressed.